src/libsyntax/parse/parser/diagnostics.rs

```diff
@@ -326,34 +326,8 @@ impl<'a> Parser<'a> {
             }
         }
 
-        let is_semi_suggestable = expected.iter().any(|t| match t {
-            TokenType::Token(token::Semi) => true, // We expect a `;` here.
-            _ => false,
-        }) && ( // A `;` would be expected before the current keyword.
-            self.token.is_keyword(kw::Break) ||
-            self.token.is_keyword(kw::Continue) ||
-            self.token.is_keyword(kw::For) ||
-            self.token.is_keyword(kw::If) ||
-            self.token.is_keyword(kw::Let) ||
-            self.token.is_keyword(kw::Loop) ||
-            self.token.is_keyword(kw::Match) ||
-            self.token.is_keyword(kw::Return) ||
-            self.token.is_keyword(kw::While)
-        );
         let sm = self.sess.source_map();
         match (sm.lookup_line(self.token.span.lo()), sm.lookup_line(sp.lo())) {
```

Comment:
`is_multiline` could be used here also?

Reply:
This block works even if any of the spans is `DUMMY_SP`, which can happen with macros. This approach is a bit more robust than `is_multiline`, but maybe it isn't worth it to protect against it this way?

Reply:
Could that robustness be integrated into `is_multiline`?

Reply:
I'll see what I can do :)